### ISSUE TITLE
Error: Wrote 2147479552 bytes instead of the required 2680846896 bytes 

### ISSUE THREAD

**Issue by aporrasval** [sage-home/sage-model#50]
I am running SAGE to turn TNG100-1 lhalo HDF5 formatted merger trees into lhalo binary format. In the input field, under "OutputFormat", I wrote lhalo_binary_output which is supposed to convert any lhalo hdf5 formatted merger trees into binary format.

When I run SAGE, I get the following error:

`Error: Wrote 2147479552 bytes instead of the required 2680846896 bytes`

The error comes from /usr/sage.c under line number 428 because "bytes_written" does not match "(ssize_t) nbytes_to_write." 

My first guess is that write() function above may not write all the bytes at once so "bytes_written" ends up writing 2147479552 bytes instead of the required 2680846896 bytes. I tried writing some lines that tried to account for this partial writing: 

```
ssize_t total_bytes_written = 0;
ssize_t bytes_written = write(fd, ((char*)&var) + total_bytes_written, nbytes_to_write - total_bytes_written);
total_bytes_written += bytes_written;
```

After including this, I get the same error (Error: Wrote 2147479552 bytes instead of the required 2680846896 bytes).

I also checked the size of sizeof(bytes_written) and sizeof(nbytes_to_write) and they provide the same number so their size is the same but when it comes to line 428 of /usr/sage.c, "bytes_written_to_match != (ssize_t) nbytes_to_write" is true. Clearly, there is some knowledge wholes about I don't understand in order to fix this error.

**Comment by manodeep** [comment]
This turned out to be a fairly unusual bug - there is a limit on linux on the total amount of data that can be written in one shot https://stackoverflow.com/questions/70368651/why-cant-linux-write-more-than-2147479552-bytes

**Comment by manodeep** [comment]
Fixed by #52 

@aporrasval Can you please check that if you run `sage` from the `master` branch then you do not get this error any longer?
